Tests/TensorFlowTests/TensorGroupTests.swift

```diff
@@ -22,34 +22,129 @@ extension TensorDataType : Equatable {
     var w, b: Tensor<Float>
+
+    init(w: Tensor<Float>, b: Tensor<Float>) {
+        self.w = w
+        self.b = b
+    }
+
+    init(handles: [_AnyTensorHandle]) {
+        precondition(handles.count == 2)
+        w = Tensor<Float>(handle: TensorHandle<Float>(handle: handles[0]))
+        b = Tensor<Float>(handle: TensorHandle<Float>(handle: handles[1]))
+    }
+
+    public var _tensorHandles: [_AnyTensorHandle] { [w.handle.handle, b.handle.handle] }
 }
 
 struct Mixed : TensorGroup, Equatable {
     // Mutable.
     var float: Tensor<Float>
     // Immutable.
     let int: Tensor<Int32>
+
+    init(float: Tensor<Float>, int: Tensor<Int32>) {
+        self.float = float
+        self.int = int
+    }
+
+    public init(handles: [_AnyTensorHandle]) {
+        precondition(handles.count == 2)
+        float = Tensor<Float>(handle: TensorHandle<Float>(handle: handles[0]))
+        int = Tensor<Int32>(handle: TensorHandle<Int32>(handle: handles[1]))
+    }
+
+    public var _tensorHandles: [_AnyTensorHandle] {
+        [float.handle.handle, int.handle.handle]
+    }
 }
 
 struct Nested : TensorGroup, Equatable {
     // Immutable.
     let simple: Simple
     // Mutable.
     var mixed: Mixed
+
+    init(simple: Simple, mixed: Mixed) {
+        self.simple = simple
+        self.mixed = mixed
+    }
+
+    public init(handles: [_AnyTensorHandle]) {
+        let simpleEnd = Int(Simple._tensorHandleCount)
+        simple = Simple(handles: Array(handles[0..<simpleEnd]))
+        mixed = Mixed(handles: Array(handles[simpleEnd..<handles.count]))
+    }
+
+    public var _tensorHandles: [_AnyTensorHandle] {
+        simple._tensorHandles + mixed._tensorHandles
+    }
 }
 
 struct Generic<T: TensorGroup & Equatable, U: TensorGroup & Equatable> : TensorGroup, Equatable {
     var t: T
     var u: U
+
+    public init(t: T, u: U) {
+        self.t = t
+        self.u = u
+    }
+
+    public init(handles: [_AnyTensorHandle]) {
+        let tEnd = Int(T._tensorHandleCount)
+        t = T.init(handles: Array(handles[0..<tEnd]))
+        u = U.init(handles: Array(handles[tEnd..<handles.count]))
+    }
+
+    public var _tensorHandles: [_AnyTensorHandle] {
+        t._tensorHandles + u._tensorHandles
+    }
+}
+
+struct UltraNested<T: TensorGroup & Equatable, V: TensorGroup & Equatable>
+    : TensorGroup, Equatable {
+    var a: Generic<T, V>
+    var b: Generic<V, T>
+
+    init(a: Generic<T, V>, b: Generic<V,T>) {
+        self.a = a
+        self.b = b
+    }
+
+    init(handles: [_AnyTensorHandle]) {
+        let firstEnd = Int(Generic<T,V>._tensorHandleCount)
+        a = Generic<T,V>.init(
+            handles: Array(handles[0..<firstEnd]))
+        b = Generic<V,T>.init(
+            handles: Array(handles[firstEnd..<handles.count]))
+    }
+
+    public var _tensorHandles: [_AnyTensorHandle] {
+        return a._tensorHandles + b._tensorHandles
+    }
+}
+
+func copyOf<T>(handle: TensorHandle<T>) -> _AnyTensorHandle {
```

Comment:
Random thoughts (since this is a test, naming doesn't matter at all!): 

Swift naming conventions recommend that when a verb phrase that the function represents ("copy of handle") contains a prepositional phrase formed with an argument ("of handle"), make the preposition be an argument label. Also, when an argument label merely repeats the type information of the argument, it ("handle") should be omitted.

What do you think of the following?

```suggestion
func copy<T>(of handle: TensorHandle<T>) -> _AnyTensorHandle {
```

Reply:
Done.